### PR TITLE
[SPIR-V] Align memory scopes with SPIR-V specification and LLVM expectations and change default mem scope value

### PR DESCRIPTION
**Pull request by VyacheslavLevytskyy**
This PR is an attempt to align memory scopes with SPIR-V specification and LLVM requirements/expectations with respect to usage of "singlethread" / "" memory scopes. It also changes default mem scope value to the most conservative/safe choice that is the System.

This PR is to partially implement ideas and code review suggestions from https://github.com/llvm/llvm-project/pull/106429 and if this PR (https://github.com/llvm/llvm-project/pull/108528) is to be merged, the merge must be postponed until after Clang-related changes from https://github.com/llvm/llvm-project/pull/106429 are merged to keep OpenCL default mem scope intact (see clang/lib/CodeGen/CGAtomic.cpp changes https://github.com/llvm/llvm-project/pull/106429 and https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_C.html#atomic-functions for definition of default mem scope in the OpenCL spec).

See https://github.com/llvm/llvm-project/pull/106429 for the detailed discussion.

### CODE REVIEW

**Comment by VyacheslavLevytskyy** [comment]
@AlexVlx @arsenm @michalpaszkowski @MrSidims What do you think about this small and clear change as a way to resolve terminology and default portions of our discussion from https://github.com/llvm/llvm-project/pull/106429 ?

I don't update any test yet until I know your opinion.

**Comment by AlexVlx** [comment]
This looks very tidy, many thanks for doing it!

File: llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp [hunk 1]
```diff
@@ -847,25 +852,18 @@ bool SPIRVInstructionSelector::selectBitcast(Register ResVReg,
...
-    // consumed in an OpenCL environment, we use the same approach and set the
-    // scope to memory_scope_device.
+  if (Ord == SSIDs.WorkGroupSSID)
+    return SPIRV::Scope::Workgroup;
+  if (Ord == SSIDs.DeviceSSID)
     return SPIRV::Scope::Device;
+  if (Ord == SyncScope::System)
+    return SPIRV::Scope::CrossDevice;
+  // Use the most conservative/safe choice as the default.
+  return SPIRV::Scope::CrossDevice;
```
Comment (arsenm) [inline llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp:866]: I'd expect this to also be caught by an explicit Ord == SyncScope::System above where the other generic ID is handled

**Comment by VyacheslavLevytskyy** [comment]
I close this, because https://github.com/llvm/llvm-project/pull/106429 got integrated suggestions from this PR

**Pull request closed without merging**